hail/src/main/scala/is/hail/expr/ir/functions/BlockMatrixEntries.scala

```diff
@@ -0,0 +1,15 @@
+package is.hail.expr.ir.functions
+import is.hail.HailContext
+import is.hail.expr.ir.TableValue
+import is.hail.expr.types.virtual.{TFloat64Optional, TInt64Optional, TStruct}
+import is.hail.expr.types.{BlockMatrixType, TableType}
+import is.hail.linalg.BlockMatrix
+
+case class BlockMatrixEntries() extends BlockMatrixToTableFunction {
```

Comment:
I have a sense that this shouldn't be hidden inside a function -- it seems like going from BM to entries table may be somewhat common, and there are plenty of optimizations we can apply (like pushing filters inside the BM or the BMEntries).

Are there other BMToTable functions planned? I think this is a very reasonable piece of infrastructure, I just may prefer to see BlockMatrixEntries() as a TableIR node.

Reply:
Hm that's fair. I'm not aware of other functions that would go from BlockMatrix -> Table

Reply:
I'll turn it into a TableIR node, seems much more straightforward and I can't think of what other functions we'd create.